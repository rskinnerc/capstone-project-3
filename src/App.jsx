import { Route, Routes } from 'react-router-dom';
import Home from './pages/Home';
import Genre from './pages/Genre';
import './App.css';
import Navbar from './components/Navbar';

function App() {
  return (
    <div>
      <Navbar />
      <Routes>
        <Route index element={<Home />} />
        <Route path="/genre/:genreSlug" element={<Genre />} />
      </Routes>
    </div>
  );
}

export default App;
