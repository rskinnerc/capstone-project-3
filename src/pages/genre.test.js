import { screen, render } from "@testing-library/react";
import store from '../redux/configureStore'
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import App from "../App"

describe("The home page functionality", () => {
  it("should load the games for a given genre from the API and list them from Redux store", async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/genre/action']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    );

    const gamesBreakdownText = await screen.findByText("GAMES BREAKDOWN", { exact: false });
    const votesText = await screen.findAllByText("votes", { exact: false });
    expect(gamesBreakdownText).toBeInTheDocument();
    expect(votesText.length).toBeGreaterThan(0);
  })

  it("should have the heading containing the games count for a given genre", async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={['/genre/action']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    );

    const gamesCountText = await screen.findByText("Games Count:", { exact: false });
    expect(gamesCountText).toBeInTheDocument();
  })
});