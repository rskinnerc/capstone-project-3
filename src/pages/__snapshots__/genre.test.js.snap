// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The home page functionality should maintain th sam snapshot between renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="flex flex-row justify-around justify-items-center bg-marino-600 text-white py-2"
        >
          <h1
            class="basis-10/12 pl-3 font-semibold font-gill"
          >
            <a
              href="/"
            >
              <i
                class="bi bi-caret-left-fill"
                data-testid="backButton"
              />
            </a>
             
            The Games Database
          </h1>
          <i
            class="bi bi-mic-fill justify-self-end basis-1/12"
          />
          <i
            class="bi bi-gear-fill justify-self-end basis-1/12"
          />
        </nav>
        <div
          class="h-52 flex flex-col justify-end font-lato bg-contain"
          style="background-image: url(https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg);"
        >
          <h2
            class="text-3xl text-right px-3 bg-marino-500/80 text-white font-bold"
          >
            Action
             
            Games
          </h2>
          <span
            class="px-3 text-right bg-marino-500/80 text-white"
          >
            Games Count:
             
            153243
          </span>
        </div>
        <div
          class="font-lato text-sm bg-marino-600 text-white p-1 uppercase"
        >
          Action GAMES BREAKDOWN
        </div>
        <ul>
          <li
            class="flex justify-between font-gill odd:bg-marino-500 even:bg-marino-700 text-white items-center px-2 py-6"
          >
            <span
              class="font-bold"
            >
              Grand Theft Auto V
            </span>
            <div
              class="flex flex-col items-center"
            >
              <span>
                4.48 / 5
              </span>
              <span
                class="text-marino-300"
              >
                5720 votes
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="flex flex-row justify-around justify-items-center bg-marino-600 text-white py-2"
      >
        <h1
          class="basis-10/12 pl-3 font-semibold font-gill"
        >
          <a
            href="/"
          >
            <i
              class="bi bi-caret-left-fill"
              data-testid="backButton"
            />
          </a>
           
          The Games Database
        </h1>
        <i
          class="bi bi-mic-fill justify-self-end basis-1/12"
        />
        <i
          class="bi bi-gear-fill justify-self-end basis-1/12"
        />
      </nav>
      <div
        class="h-52 flex flex-col justify-end font-lato bg-contain"
        style="background-image: url(https://media.rawg.io/media/games/34b/34b1f1850a1c06fd971bc6ab3ac0ce0e.jpg);"
      >
        <h2
          class="text-3xl text-right px-3 bg-marino-500/80 text-white font-bold"
        >
          Action
           
          Games
        </h2>
        <span
          class="px-3 text-right bg-marino-500/80 text-white"
        >
          Games Count:
           
          153243
        </span>
      </div>
      <div
        class="font-lato text-sm bg-marino-600 text-white p-1 uppercase"
      >
        Action GAMES BREAKDOWN
      </div>
      <ul>
        <li
          class="flex justify-between font-gill odd:bg-marino-500 even:bg-marino-700 text-white items-center px-2 py-6"
        >
          <span
            class="font-bold"
          >
            Grand Theft Auto V
          </span>
          <div
            class="flex flex-col items-center"
          >
            <span>
              4.48 / 5
            </span>
            <span
              class="text-marino-300"
            >
              5720 votes
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
